app/models/case_log.rb

```diff
@@ -718,4 +723,8 @@ def soft_value_for_period(value)
 
     (value * 52 / num_of_weeks).round(2)
   end
+
+  def upcase_and_remove_whitespace(string)
+    string.present? ? string.upcase.gsub(/\s+/, "") : string
```

Comment:
`string.upcase.gsub(/\s+/, "") || string` should do the trick too

Reply:
I like it but this wouldn't protect a call made with a nil value.